From 7a7cc97a5d5f0ffc463df30f36922dfeb2ac8fbd Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:20:37 -0400
Subject: [fs] nfsd: Remove nfs4_lock_state(): nfsd4_release_lockowner

Message-id: <1410898849-22126-154-git-send-email-bfields@redhat.com>
Patchwork-id: 95739
O-Subject: [Patch RHEL7 153/165] nfsd: Remove nfs4_lock_state(): nfsd4_release_lockowner
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 51f5e78355db2e9b4d5d9093f83be3567178d236
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 712d7e7..0f9e842 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -5577,11 +5577,9 @@ nfsd4_release_lockowner(struct svc_rqst *rqstp,
  dprintk("nfsd4_release_lockowner clientid: (%08x/%08x):\n",
   clid->cl_boot, clid->cl_id);
 
- nfs4_lock_state();
-
  status = lookup_clientid(clid, cstate, nn);
  if (status)
-  goto out;
+  return status;
 
  clp = cstate->clp;
  /* Find the matching lock stateowner */
@@ -5598,7 +5596,7 @@ nfsd4_release_lockowner(struct svc_rqst *rqstp,
    if (check_for_locks(stp->st_stid.sc_file, lo)) {
     status = nfserr_locks_held;
     spin_unlock(&clp->cl_lock);
-    goto out;
+    return status;
    }
   }
 
@@ -5608,8 +5606,6 @@ nfsd4_release_lockowner(struct svc_rqst *rqstp,
  spin_unlock(&clp->cl_lock);
  if (lo)
   release_lockowner(lo);
-out:
- nfs4_unlock_state();
  return status;
 }
 
-- 
1.7.1